scalafix-sbt/src/main/scala/scalafix/internal/sbt/JGitCompletions.scala

```diff
@@ -0,0 +1,32 @@
+package scalafix.internal.sbt
+
+import java.nio.file.Path
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.lib.RefDatabase
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.util.GitDateFormatter
+import scala.collection.JavaConverters._
+
+class JGitCompletion(cwd: Path) {
+  val builder = new FileRepositoryBuilder()
+  val repo = builder.readEnvironment().setWorkTree(cwd.toFile).build()
+  val refList = repo.getRefDatabase().getRefs(RefDatabase.ALL).asScala
+  val branchAndTags = refList.map {
+    case (a, b) => Repository.shortenRefName(a)
+  }
+  val git = new Git(repo)
+  val refs = git.log().setMaxCount(20).call().asScala.toList
+  val dateFormatter = new GitDateFormatter(GitDateFormatter.Format.RELATIVE)
+
+  val last20Commits = refs.map { ref =>
+    val relativeCommitTime =
+      dateFormatter.formatDate(refs.head.getCommitterIdent)
+    val abrev = ref.abbreviate(8).name
+    val short = ref.getShortMessage
+    // TODO: figure out how to display the whole message but only auto-complete on abrev
+    s"$abrev -- $short ($relativeCommitTime)"
+  }
+
+  def all: Iterable[String] = branchAndTags ++ last20Commits
```

Comment:
This simulates the native `git diff <tab>`

Reply:
`> git diff <tab>`

```
ag/gh-pages                         me/drop-in                          olaf/explicit-implicit-args         stash                             
ag/issue-260                        me/drop-in-jgit                     olaf/explicit-return                team/gh-pages                     
ag/issue-319                        me/feature/368-escape-patches       olaf/fail-fast-264                  team/HEAD                         
ag/master                           me/feature/495-custom-id            olaf/fileline                       team/master                       
bugfix-publishSigned                me/feature/508-escape-diff          olaf/gh-pages                       team/publish-setupt-doc           
bugfix-publishSigned2               me/feature/518-improve-sbt-parser   olaf/hijack-reporter                teskit-empty-assertdiff           
bugfix-sbt-1.0                      me/gh-pages                         olaf/implicit-args                  testkit-multiline-asserts         
contributing                        me/intellij                         olaf/imports                        trim_includeNewLine               
custom-message                      me/master                           olaf/intellij-buidl                 unify-config-style                
debug-publish                       me/multi-scala-resources            olaf/logger2                        v0.0.1                            
diff-unkown-hash                    me/no-finalize                      olaf/master                         v0.1.0                            
disable-keywords                    me/publish-setupt-doc               olaf/metadoc                        v0.2.0                            
disable-range                       me/scastie-like-build               olaf/mirror                         v0.2.0-RC1                        
disable-regex                       me/semantic                         olaf/mirror-constructor             v0.2.1                            
disable-sbt-1.0                     me/teskit-empty-assertdiff          olaf/no-auto-apply                  v0.2.2                            
disable-syntax                      me/testkit-multiline-asserts        olaf/no-delayed-init                v0.2.2-RC1                        
document-disable-syntax             me/unify-config-style               olaf/no-postfix                     v0.2.2-RC2                        
drop-in                             more-custom-message                 olaf/note                           v0.2.2-RC3                        
drop-in-jgit                        more-custom-message2                olaf/no-tparams                     v0.3.0                            
feature/368-escape-patches          multi-scala-resources               olaf/organize-imports2              v0.3.1                            
feature/495-custom-id               nl                                  olaf/partialN-to-functionN          v0.3.2                            
feature/508-escape-diff             no-finalize                         olaf/patch3                         v0.3.3                            
feature/512-avoid-stale-semanticdb  olaf/0.3.4                          olaf/release                        v0.3.4                            
feature/518-improve-sbt-parser      olaf/110                            olaf/rename                         v0.4.0                            
FETCH_HEAD                          olaf/189                            olaf/safetcheck                     v0.4.1                            
HEAD                                olaf/2.12.4                         olaf/sbt1                           v0.4.2         
issue-260                           olaf/252                            olaf/sbt-scalafix-haul              v0.5.0                            
marcelo/gh-pages                    olaf/254                            olaf/scastie-like-build             v0.5.0-M1                         
marcelo/master                      olaf/255                            olaf/significant-indent             v0.5.0-M2                         
marcelo/trim                        olaf/807                            olaf/store-reporter                 v0.5.0-M3                         
marcelo/trim_includeNewLine         olaf/acyclic                        olaf/tab                            v0.5.0-M4                         
marcelo/trim_whitespaces            olaf/autotupl                       olaf/team                           v0.5.0-M5                         
master                              olaf/cache                          olaf/termdisplay                    v0.5.0-M6                         
me/bugfix-publishSigned             olaf/classifiers                    olaf/testkit                        v0.5.0-M7                         
me/bugfix-publishSigned2            olaf/classpath=auto                 olaf/the-big-cleanup                v0.5.0-M8                         
me/bugfix-sbt-1.0                   olaf/column                         olaf/trailing-comma                 v0.5.0-RC1                        
me/custom-message                   olaf/commands                       olaf/upgrade-meta                   v0.5.0-RC2                        
me/debug-publish                    olaf/compiler-messages              olaf/utest                          v0.5.0-RC3                        
me/diff-unkown-hash                 olaf/config                         olaf/xml-lit-to-xml-interpolator    v0.5.1                            
me/disable-keywords                 olaf/contributing                   olaf/xor2eitehr                     v0.5.2                            
me/disable-range                    olaf/coursier                       ORIG_HEAD                           v0.5.3                            
me/disable-regex                    olaf/delayed-init                   publish-setupt-doc                  v0.5.4                            
me/disable-sbt-1.0                  olaf/dottyhost                      release-process                     v0.5.5                            
me/disable-syntax                   olaf/expand-sugars                  sbt1                                v0.5.6                            
me/document-disable-syntax          olaf/explicitimplicit               scastie-like-build                                                    
51abefa0  -- [HEAD]    wip improve sbt completion (16 minutes ago)
3f63b5bf  -- [HEAD^]   Merge pull request #474 from marcelocenerine/trim_includeNewLine (7 hours ago)
47b94c03  -- [47b94c03] Add leadingSpaces/trailingSpaces methods to TokenList (8 hours ago)
8bd026de  -- [HEAD^^]  Merge pull request #503 from MasseGuillaume/feature/368-escape-patches (4 days ago)
20e445fe  -- [20e445fe] Escape hatch on Patch (fix #368) (4 days ago)
a2c5d70f  -- [HEAD~3]  Merge pull request #500 from MasseGuillaume/feature/495-custom-id (4 days ago)
59efe7df  -- [59efe7df] Add CustomMessage to the public api (4 days ago)
9ae60718  -- [9ae60718] Add id for CustomMessage (fix #495) (7 days ago)
e4a5c35d  -- [HEAD~4]  Merge pull request #494 from MasseGuillaume/disable-regex (7 days ago)
a4228608  -- [HEAD~5]  Merge pull request #497 from olafurpg/disable-signatures (7 days ago)
79309479  -- [79309479] DisableSyntax add regex (7 days ago)
5dbdd6bd  -- [5dbdd6bd] IntervalSet test for empty and add toString (7 days ago)
b022fbd4  -- [b022fbd4] DisableSyntax don't repeat DisableSyntax.keyword in message (7 days ago)
a992b02f  -- [a992b02f] Assert instead of scalafix:ok (8 days ago)
7896ccd2  -- [7896ccd2] Refactor Disable to use views. (8 days ago)
58acdbeb  -- [58acdbeb] Fix #493, handle synthetics and symbol signatures in Disable. (8 days ago)
b48d7f0c  -- [HEAD~6]  Merge pull request #490 from olafurpg/unmanagedSources (9 days ago)
e9b2b0ab  -- [e9b2b0ab] s/canFormat/canFix/ (12 days ago)
26be6fa5  -- [26be6fa5] Use unmanagedSources instead of unmanagedSourceDirectories. (12 days ago)
4d46001d  -- [HEAD~7]  Merge pull request #488 from olafurpg/master (12 days ago)
```